mk/stack.mk

```diff
@@ -1,4 +1,7 @@
-STACK=stack
+# Andreas, 2022-03-10: suppress chatty announcements like
+# "Stack has not been tested with GHC versions above 9.0, and using 9.2.2, this may fail".
+# These might get in the way of interaction testing.
+STACK=stack --silent
```

Comment:
The `--silent` option also suppress messages about the compilation process like

```
Building library for Agda-2.6.3..
[  1 of 406] Compiling Agda.Interaction.ExitCode
[  2 of 406] Compiling Agda.Termination.CutOff
[  3 of 406] Compiling Agda.Termination.Semiring
[  4 of 406] Compiling Agda.TypeChecking.Monad.Debug[boot]
[  5 of 406] Compiling Agda.Utils.AffineHole
[  6 of 406] Compiling Agda.Utils.Applicative
[  7 of 406] Compiling Agda.Utils.CallStack.Base
[  8 of 406] Compiling Agda.Utils.Char
[  9 of 406] Compiling Agda.Utils.Environment
[ 10 of 406] Compiling Agda.Utils.Fail
[ 11 of 406] Compiling Agda.Utils.Function
[ 12 of 406] Compiling Agda.TypeChecking.SizedTypes.Utils
```

Without these messages we haven't any feedback from the compilation from ~40 minutes (and I don't   care the `Stack has not been tested with Cabal versions above 3.4, but version 3.6.3.0 was found, this may fail` message).

Reply:
Ok, maybe we need to adjust, to only pass `--silent` when the interaction testsuite is run (or another testsuite that captures the output).

Reply:
See #5829.